src/analyses/variable-sensitivity/abstract_enviroment.cpp

```diff
@@ -191,7 +191,7 @@ bool abstract_environmentt::assign(
   {
     INVARIANT(s.id()==ID_symbol, "Have a symbol or a stack");
     const symbol_exprt &symbol_expr(to_symbol_expr(s));
-    if(map.find(symbol_expr)==map.end())
+    if(!map.has_key(symbol_expr))
```

Comment:
Below (line 201) we can use `find(symbol_expr, tvt(true))` instead of `[]`. The second parameter tells it that we know that `symbol_expr` is definitely in the map (since we checked `has_key()` already). This avoids traversing the path from the root to the element twice (once for checking whether the element is in the map, and another time for detaching nodes).

Also we should move the whole code portion (lines 190-203) into the then-branch of the subsequent if (lines 211-215). Currently the map is modified in line 201 even if `lhs_value` is not modified later on.

In line 239 we should also use the const version of `find()` to retrieve the item first. Then, do the comparison with `final_value`. Then if we found that `symbol_expr` was in the map and the value was not equal to `final_value`, we can use the non-const `find(symbol_expr, tvt(true)` to retrieve a non-const reference to the value and assign the `final_value` to that.